Duplicates in ORT license-classifications.yml generated file make ORT fail

The license-classifications.yml file, as [advertised in ORT's repo](https://github.com/oss-review-toolkit/ort/blob/a531ccb99e66d8bd035867524c52a5033a47b63c/docs/config-file-license-classifications-yml.md) has a few duplicates that make ORT fail on the evalutor phase. 

Full exception is reproduced below:

```
12:51:48.726 [main] INFO  org.ossreviewtoolkit.cli.commands.EvaluatorCommand - Read ORT result from 'advisor-result.json' (3.35 MiB) in 1.405892857s.
Exception in thread "main" com.fasterxml.jackson.databind.exc.ValueInstantiationException: Cannot construct instance of `org.ossreviewtoolkit.model.licenses.LicenseClassifications`, problem: Found multiple license categorizations with the same id: [LGPL-3.0-only, AGPL-3.0-only, ICU, GPL-3.0-only, PSF-2.0]
 at [Source: (File); line: 3264, column: 1]
	at com.fasterxml.jackson.databind.exc.ValueInstantiationException.from(ValueInstantiationException.java:47)
	at com.fasterxml.jackson.databind.DeserializationContext.instantiationException(DeserializationContext.java:2047)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.wrapAsJsonMappingException(StdValueInstantiator.java:587)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.rewrapCtorProblem(StdValueInstantiator.java:610)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:293)
	at com.fasterxml.jackson.module.kotlin.KotlinValueInstantiator.createFromObjectWith(KotlinValueInstantiator.kt:125)
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:202)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:444)
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1405)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:352)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185)
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:323)
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4674)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3514)
	at org.ossreviewtoolkit.cli.commands.EvaluatorCommand.run(EvaluatorCommand.kt:804)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:198)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:211)
	at com.github.ajalt.clikt.parsers.Parser.parse(Parser.kt:18)
	at com.github.ajalt.clikt.core.CliktCommand.parse(CliktCommand.kt:400)
	at com.github.ajalt.clikt.core.CliktCommand.parse$default(CliktCommand.kt:397)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:415)
	at com.github.ajalt.clikt.core.CliktCommand.main(CliktCommand.kt:440)
	at org.ossreviewtoolkit.cli.OrtMainKt.main(OrtMain.kt:83)
Caused by: java.lang.IllegalArgumentException: Found multiple license categorizations with the same id: [LGPL-3.0-only, AGPL-3.0-only, ICU, GPL-3.0-only, PSF-2.0]
	at org.ossreviewtoolkit.model.licenses.LicenseClassifications.<init>(LicenseClassifications.kt:86)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:128)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:291)
	... 18 more
```

I've fixed (removed) these duplicates and am willing to submit a PR, however I understand that this file is generated.  

Does it make any sense to include a fix to the generated file? At least people coming from ORT's documentation would not struggle with the same issue.

--- a/src/Generators/OrtLicenseClassificationYml.hs
+++ b/src/Generators/OrtLicenseClassificationYml.hs
@@ -1,7 +1,8 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE LambdaCase #-}
 module Generators.OrtLicenseClassificationYml
-    ( writeOrtLicenseClassificationYml
+    ( generateOrtLicenseClassification
+    , writeOrtLicenseClassificationYml
     ) where
 
 
@@ -11,6 +12,8 @@ import           MyPrelude
 import qualified Data.ByteString.Lazy as BL
 import           Data.Aeson.Encode.Pretty (encodePretty)
 import qualified Data.Yaml as Y
+import           Data.List (nub, groupBy, sortBy)
+import           Data.Function (on)
 
 import           Model.License
 import           Processors.ToPage (Page (..), LicenseDetails (..), unpackWithSource)
@@ -121,8 +124,8 @@ isOsiApprovedCategory = OLCY_Category "osi-approved" Nothing
 isFSFFreeCategory = OLCY_Category "fsf-free" Nothing
 
 ratingToCategoryName :: Rating -> [OLCY_Category_Name]
-ratingToCategoryName (RUnknown rs) = map (\r -> "maybe-rating:" ++ (show r)) rs
-ratingToCategoryName r             = ["rating:" ++ (show r)]
+ratingToCategoryName (RUnknown rs) = map (\r -> "maybe-rating:" ++ show r) rs
+ratingToCategoryName r             = ["rating:" ++ show r]
 ratingToCategory :: Rating -> [OLCY_Category]
 ratingToCategory r = map (`OLCY_Category` Nothing) (ratingToCategoryName r)
 
@@ -199,11 +202,20 @@ pageToCategorization page = let
   ++ categorizationsFromFSFFree
   )
 
+generateOrtLicenseClassification :: [Page] -> OrtLicenseClassificationYml
+generateOrtLicenseClassification  pages = let
+  categorizations = (
+    map (foldl1 (\(OLCY_Categorization id1 cs1) (OLCY_Categorization _ cs2) ->
+                   OLCY_Categorization id1 (nub (cs1 ++ cs2)))) .
+    groupBy ((==) `on`_OLCY_Category_id).
+    sortBy (compare `on` _OLCY_Category_id) .
+    map pageToCategorization) pages
+  allCategories = nub (categories ++ concatMap _OLCY_Category_categories categorizations)
+  in OrtLicenseClassificationYml allCategories categorizations
+
 writeOrtLicenseClassificationYml :: FilePath -> [Page] -> IO ()
 writeOrtLicenseClassificationYml  outputFolder pages = let
-  categorizations = map pageToCategorization pages
-  allCategories = nub (categories ++ concatMap _OLCY_Category_categories categorizations)
-  ortLicenseClassificationYml = OrtLicenseClassificationYml allCategories categorizations
+  ortLicenseClassificationYml = generateOrtLicenseClassification pages
   in do
     createDirectoryIfNotExists (outputFolder </> "ort")
     Y.encodeFile (outputFolder </> "ort" </> "license-classifications.yml") ortLicenseClassificationYml

